Included Export CSV for selection of Tags

swang1997 opened the pull request.

swang1997 committed:
Stats function included

swang1997 committed:
Update Stats

swang1997 committed:
Update Stats.java

swang1997 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F11-1/main

swang1997 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F11-1/main

swang1997 committed:
Changed StatsCommand

swang1997 committed:
Included student info while exporting

swang1997 committed:
Changed csv location

swang1997 committed:
Included Selection of Tag for Exporting CSV

swang1997 committed:
Updated CSV for selection

swang1997 committed:
merge conflict

swang1997 committed:
pass checkstyle

Tejasvas123 commented on export.csv:
```diff
@@ -0,0 +1,5 @@
+Locker,Address,Zone,Status,Name,Matrix Number,Course,Email
```
add this to gitignore

swang1997 replied:
added

Tejasvas123 commented on export_selected.csv:
```diff
@@ -0,0 +1,5 @@
+Locker,Zone,Status,Name
```
same as above

swang1997 replied:
removed export_selected.csv. Both exports will share same csv file.

Tejasvas123 commented on src/main/java/duke/logic/commands/ExportLockerSelectCommand.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public void execute(LockerList lockerList, Ui ui, Storage storage) throws DukeException {
+        ui.exportSelect();
```
why are you passing ui.readCommand()?

Tejasvas123 replied:
the read command is already being executed in duke.java

swang1997 replied:
edited in parser

Tejasvas123 commented on src/main/java/duke/parser/ExportLockerSelectCommandParser.java:
```diff
@@ -0,0 +1,15 @@
+
+    /**
+     * This function is used to parse the user input for exporting the details as a CSV file.
+     * @param arguments this is the user input string for the tags that the user wants
```
the actual parsing should be done here. Like basically in your case you should parse what all fields are being included into export. Have a look at some examples like AddCommandParser/EditCommandParser to understand what i mean

swang1997 replied:
edited

Tejasvas123 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -24,11 +24,15 @@
+    private static final String LIST_COMMAND = "list";
+    private static final String EXIT_COMMAND = "bye";
+    private static final String EXPORT_LOCKER_COMMAND = "export";
+    private static final String EXPORT_SELECT_COMMAND = "exports";
```
if i am not wrong i removed all this in my latest PR. The one about code cleaning

swang1997 replied:
removed and cleaned

Tejasvas123 commented on src/main/java/duke/storage/ExportCsv.java:
```diff
@@ -49,7 +49,8 @@ public static void exportLockers(List<Locker> lockerList) throws DukeException {
+                    
                     details[FIFTH_COLUMN] = l.getUsage().get().getStudent().getName().getName();
                     details[SIXTH_COLUMN] = l.getUsage().get().getStudent().getMatricNumber().getMatricId();
                     details[SEVENTH_COLUMN] = l.getUsage().get().getStudent().getMajor().getCourse();
```
how about exporting rental dates also? or like no-need for that?

swang1997 replied:
will try doing it....

Tejasvas123 commented on src/main/java/duke/storage/ExportSelection.java:
```diff
@@ -0,0 +1,125 @@
+import java.util.List;
+
+public class ExportSelection {
+    private static final String CSV_OUTPUT_PATH = "export_selected.csv";
```
i was thinking of using only one file.  Like just one export.csv

swang1997 replied:
fixed

Tejasvas123 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -13,6 +13,9 @@ public String readCommand() {
         return sc.nextLine();
     }
 
+    public String read() {
```
why is this needed?

swang1997 replied:
removed

Tejasvas123 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -129,6 +132,30 @@ public void showNoAvailableLockers() {
         printSpaces(" There are no available lockers at the moment. ");
         printSpaces(" Unfortunately I will have to terminate the subscription of the student");
     }
+    
```
remove this snoozetask function

swang1997 replied:
removed

Tejasvas123 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -193,6 +220,30 @@ public void exportMessage() {
+    }
+
+
+    /**
```
remove functions keywordPrint and remindersList

swang1997 replied:
removed

swang1997 committed:
clean code

swang1997 committed:
Fixed Minor Issues of Export Selection of CSV, adjusted export selection of parser files.

Fixed Minor Issues of Export Selection of CSV, adjusted export selection of parser files.

swang1997 committed:
added in case check

swang1997 committed:
included start and end date for both methods of exporting csv file.

swang1997 committed:
fixed checkstyle

swang1997 committed:
minor fixes to parser export

swang1997 committed:
Modified input for Export CSV selection

Tejasvas123 commented on src/main/java/duke/parser/ExportLockerSelectCommandParser.java:
```diff
@@ -0,0 +1,70 @@
+                    if (str.equals("name")) {
+                        throw new DukeException(ExportLockerSelectCommand.MISSINGSTATUS_FORMAT);
+                    }
+                    if (str.equals("matrix")) {
```
should be matricid /student id?

Tejasvas123 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -1,5 +1,6 @@
 package duke.storage;
 
+import duke.Duke;
```
unused class?

Tejasvas123 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -193,6 +192,14 @@ public void exportMessage() {
+     * This function tells the user that SpongeBob has exported the CSV file.
+     */
+    public void exportSelect() {
+        printSpaces("Noted. I have exported the selected details to a new CSV file");
```
just write the name of the file instead of writing csv files

Tejasvas123 replied:
does it work as intended? like no errors?

Tejasvas123 reviewed (commented):
Suggested some minor changes. Great job otherwise!

Tejasvas123 commented on src/main/java/duke/storage/StorageManager.java:
```diff
@@ -32,4 +37,9 @@ public void exportAsCsv(LockerList lockerList) throws DukeException {
         writeToCsv.exportLockers(lockerList.getAllLockers());
     }
+
+    public void exportSelection(LockerList lockerList, String input) throws DukeException {
```
forgot to add override?

swang1997 committed:
Minor Adjustments

swang1997 committed:
Merge branch 'master' into master

swang1997 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-F11-1/main

swang1997 committed:
Minor changes to code

swang1997 committed:
Merge branch 'master' of https://github.com/swang1997/main

Tejasvas123 reviewed (approved):
Approved! Good job!
